packages/supplementary-contracts/script/tokenunlocking/Vest.s.sol

```diff
@@ -0,0 +1,55 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.24;
+
+import "forge-std/src/Script.sol";
+import "forge-std/src/console2.sol";
+
+import "@openzeppelin/contracts/proxy/ERC1967/ERC1967Proxy.sol";
+import "@openzeppelin/contracts/token/ERC20/ERC20.sol";
+
+import "../../contracts/tokenunlocking/TokenUnlocking.sol";
+
+contract VestTokenUnlocking is Script {
+    using stdJson for string;
+
+    struct VestingItem {
+        bytes32 name; // Conversion from json "string" to bytes32 will take place in foundry,
+            // cannot use string here, as json parser cannot interpret string from json, everything
+            // is bytes-chunks. It is more of informational to script executor anyways.
+        address recipient;
+        address proxy;
+        uint256 vestAmount;
+    }
+
+    uint256 public PRIVATE_KEY = vm.envUint("PRIVATE_KEY"); // owner
+    ERC20 private tko = ERC20(vm.envAddress("TAIKO_TOKEN"));
+
+    function run() external {
+        string memory path = "/script/tokenvesting/Vest.data.json";
+        VestingItem[] memory items = abi.decode(
+            vm.parseJson(vm.readFile(string.concat(vm.projectRoot(), path))), (VestingItem[])
+        );
+
+        for (uint256 i; i < items.length; i++) {
+            address proxy = items[i].proxy;
+            console2.logBytes32(items[i].name);
+            console2.log("Grantee unlocking contract address:", proxy);
+            console2.log("Vest amount (TKO):", items[i].vestAmount);
+
+            require(TokenUnlocking(proxy).owner() == msg.sender, "msg.sender not owner");
+            require(
+                TokenUnlocking(proxy).recipient() == items[i].recipient, "inconsistent recipient"
+            );
+
+            uint128 vestAmount = uint128(items[i].vestAmount * 1e18);
+            require(tko.balanceOf(msg.sender) >= vestAmount, "insufficient TKO balance");
+
+            vm.startBroadcast(PRIVATE_KEY);
```

Comment:
We can just simply use `vm.startBroadcast()` at the beginning, since no need to have `PRIVATE_KEY` in env var, as we need to call the `forge script` with `--private-key`

Reply:
feel free to merge and modify as you wish, Dani.